examples/peer-and-content-routing/2.js

```diff
@@ -1,85 +1,59 @@
 /* eslint-disable no-console */
 'use strict'
 
-const libp2p = require('../../')
+const Libp2p = require('../../')
 const TCP = require('libp2p-tcp')
 const Mplex = require('libp2p-mplex')
 const SECIO = require('libp2p-secio')
 const PeerInfo = require('peer-info')
 const CID = require('cids')
 const KadDHT = require('libp2p-kad-dht')
-const defaultsDeep = require('@nodeutils/defaults-deep')
-const waterfall = require('async/waterfall')
-const parallel = require('async/parallel')
 
-class MyBundle extends libp2p {
-  constructor (_options) {
-    const defaults = {
-      modules: {
-        transport: [ TCP ],
-        streamMuxer: [ Mplex ],
-        connEncryption: [ SECIO ],
-        // we add the DHT module that will enable Peer and Content Routing
-        dht: KadDHT
-      },
-      config: {
-        dht: {
-          enabled: true,
-          kBucketSize: 20
-        }
+const all = require('it-all')
+const delay = require('delay')
+
+const createNode = async () => {
+  const peerInfo = await PeerInfo.create()
+  peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
+
+  const node = await Libp2p.create({
+    peerInfo,
+    modules: {
+      transport: [TCP],
+      streamMuxer: [Mplex],
+      connEncryption: [SECIO],
+      dht: KadDHT
+    },
+    config: {
+      dht: {
+        enabled: true
       }
     }
+  })
 
-    super(defaultsDeep(_options, defaults))
-  }
-}
-
-function createNode (callback) {
-  let node
-
-  waterfall([
-    (cb) => PeerInfo.create(cb),
-    (peerInfo, cb) => {
-      peerInfo.multiaddrs.add('/ip4/0.0.0.0/tcp/0')
-      node = new MyBundle({
-        peerInfo
-      })
-      node.start(cb)
-    }
-  ], (err) => callback(err, node))
+  await node.start()
+  return node
 }
 
-parallel([
-  (cb) => createNode(cb),
-  (cb) => createNode(cb),
-  (cb) => createNode(cb)
-], (err, nodes) => {
-  if (err) { throw err }
-
-  const node1 = nodes[0]
-  const node2 = nodes[1]
-  const node3 = nodes[2]
+;(async () => {
+  const [node1, node2, node3] = await Promise.all([
+    createNode(),
+    createNode(),
+    createNode()
+  ])
 
-  parallel([
-    (cb) => node1.dial(node2.peerInfo, cb),
-    (cb) => node2.dial(node3.peerInfo, cb),
-    // Set up of the cons might take time
-    (cb) => setTimeout(cb, 300)
-  ], (err) => {
-    if (err) { throw err }
+  await Promise.all([
+    node1.dial(node2.peerInfo),
+    node2.dial(node3.peerInfo)
+  ])
 
-    const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSySnL')
+  const cid = new CID('QmTp9VkYvnHyrqKQuFPiuZkiX9gPcqj6x5LJ1rmWuSySnL')
+  await node1.contentRouting.provide(cid)
 
-    node1.contentRouting.provide(cid, (err) => {
-      if (err) { throw err }
+  // wait for propagation
+  await delay(300)
 
-      console.log('Node %s is providing %s', node1.peerInfo.id.toB58String(), cid.toBaseEncodedString())
```

Comment:
Was this removed intentionally? If so, it should also be removed from the readme. Can you make sure the code in the readme matches what's in the js files

Reply:
I added it back, I removed unintentionally